src/tools/wasm-shell.cpp

```diff
@@ -292,7 +292,7 @@ int main(int argc, const char* argv[]) {
         builders[moduleName].swap(builder);
         modules[moduleName].swap(module);
         i++;
-        bool valid = WasmValidator().validate(*modules[moduleName], options.getFeatures());
+        bool valid = WasmValidator().validate(*modules[moduleName], options.getFeatures(*modules[moduleName]));
```

Comment:
with what I'm suggesting here, we'd replace changes like this to a tool like wasm-shell with a call to `computeFeatures(*modules[ModuleName])` at an earlier appropriate time